### PR TITLE
Do not create additional root for already rooted $refs

### PR DESCRIPTION
**Pull request by P0lip**
Linked ticket stoplightio/platform-internal#4954.

### CODE REVIEW

File: test/specs/custom-bundling-roots/custom-roots.spec.js [hunk 1]
```diff
@@ -392,6 +392,76 @@ describe("Custom bundling roots", () => {
...
+          mid: "2",
+        })
+        .reply(200, {
+          properties: {
+            id: {
+              type: "string"
+            }
+          },
+          definitions: {
+            bar: {
```
Comment (P0lip) [inline test/specs/custom-bundling-roots/custom-roots.spec.js:410]: I wasn't sure about this one. I decided to hoist it.

File: test/specs/custom-bundling-roots/custom-roots.spec.js [hunk 2]
```diff
@@ -392,6 +392,76 @@ describe("Custom bundling roots", () => {
...
+      let parser = new $RefParser();
+
+      const schema = await parser.bundle(__dirname, model, {
+        bundle: defaults.json_schema,
+      });
+
+      expect(schema).to.equal(parser.schema);
+      expect(schema).to.deep.equal({
+        definitions: {
+          Book_Bar: {
```
Comment (P0lip) [inline test/specs/custom-bundling-roots/custom-roots.spec.js:441]: It landed here.

File: test/specs/custom-bundling-roots/custom-roots.spec.js [hunk 3]
```diff
@@ -392,6 +392,76 @@ describe("Custom bundling roots", () => {
...
+            type: "string"
+          },
+          book: {
+            definitions: {
```
Comment (P0lip) [inline test/specs/custom-bundling-roots/custom-roots.spec.js:445]: Book, however, was inlined.

**Comment by P0lip** [comment]
I updated https://github.com/stoplightio/bundling-schema-test, you can see generated outputs there if you wish.
I'd suggest trying this out for other models / specs as well.

File: test/specs/custom-bundling-roots/reference/openapi-3.bundled.js [hunk 4]
```diff
@@ -14,10 +14,7 @@ module.exports = {
     "/flight/{id}": {
       parameters: [
         {
-          type: "number",
-          name: "id",
-          in: "path",
-          required: true
+          $ref: "#/components/schemas/Id"
```
Comment (P0lip) [inline test/specs/custom-bundling-roots/reference/openapi-3.bundled.js:17]: IIRC we didn't want to have a $ref here, but I don't remember why?
Or am I wrong?
TBH, I don't see a reason not to have a $ref here if it was authored in the original source.

Comment (P0lip) [reply test/specs/custom-bundling-roots/reference/openapi-3.bundled.js]: I think it was our oversight in the end.

**Review by wmhilton — APPROVED** [review]
LGTM

**Pull request merged**